src/rt/objects/region.cc

```diff
@@ -441,8 +442,121 @@ namespace rt::objects
       if (r != get_local_region() && r != cown_region)
       {
         to_collect.insert(r);
-        std::cout << "Collecting region: " << r << std::endl;
+        std::cout << "Collecting region: " << r << " with bridge: " << r->bridge << std::endl;
       }
     }
   }
+
+  void change_parent(Region* r, Region* p)
+  {
+    assert(!Region::is_ancestor(p, r));
+    r->parent = p;
+  }
+
+  void merge_regions(DynObject* src, DynObject* sink)
+  {
+    // TODO The 'sink' should have state *closed* in order to be merged
+    assert(src != nullptr);
+    assert(sink != nullptr);
+    assert(src->get_prototype() == objects::regionPrototypeObject());
+    assert(sink->get_prototype() == objects::regionPrototypeObject());
+    
+    sink->change_rc(-1);
+    src->change_rc(-1);
+
+    auto src_region = get_region(src);
+    auto sink_region = get_region(sink);
+    
+    // TODO yield error?
+    if (src_region == sink_region)
+    {
+      return;
+    }
+
+    if(src_region->parent != sink_region)
+    {
+      ui::error(
+      "Sink is not a parent of source",
+      src);
+    }
+    // Move all objects in the region, note that this includes bridge object
+    for (auto obj : src_region->objects)
+    {
+      auto r = get_region(obj);
+      std::cout << "Moving object: " << obj << " with region bridge: " << r->bridge << " to region with bridge: " << sink_region->bridge << std::endl;
+      obj->region = {sink_region};
+      sink_region->objects.insert(obj);
+      src_region->objects.erase(obj);
+    }
+    // Adjust direct subregions 
+    for (auto obj: src_region->direct_subregions)
+    {
+      auto r = get_region(obj);
+      change_parent(r, sink_region);
+      sink_region->direct_subregions.insert(obj);
+      src_region->direct_subregions.erase(obj);
+    }
+
+    sink_region->is_lrc_dirty = sink_region->is_lrc_dirty || src_region->is_lrc_dirty;
+    // Finalize dissasembly of region
+    sink_region->direct_subregions.erase(src);
+    auto old_proto = src->set_prototype(nullptr);
+    remove_reference(src, old_proto);
+    src_region->bridge = nullptr;
+    // Adjust sbrc
+    sink_region->sub_region_reference_count--;
+    sink_region->sub_region_reference_count += src_region->sub_region_reference_count;
+    // Adjust lrc, magic '2' since both sink and src lrc will already be incremented at this stage in execution 
+    if (src_region->local_reference_count > 1){
+      std::cout << src_region->local_reference_count << std::endl;
```

Comment:
```suggestion
      std::cout << src_region->local_reference_count << std::endl;
```

This print will probably not make much sense. I'd suggest just removing it.